src/fu-engine.c

```diff
@@ -7307,6 +7320,18 @@ fu_engine_update_history_device(FuEngine *self, FuDevice *dev_history, GError **
 		}
 	}
 
+	/* check nothing changed */
+	integrity_str = fwupd_release_get_metadata_item(rel_history, "SystemIntegrityOld");
+	if (integrity_str != NULL) {
+		g_autoptr(GHashTable) integrity = fu_engine_integrity_new(NULL);
+		if (integrity != NULL) {
+			g_autofree gchar *str = fu_engine_integrity_to_string(integrity);
+			fwupd_release_add_metadata_item(FWUPD_RELEASE(rel_history),
+							"SystemIntegrityNew",
+							str);
+		}
+	}
+
```

Comment:
I would argue that you should include the data for new even if old wasn't present.  It's more data points that LVFS can sift through on a given platform.